drivers/eeprom/eeprom_st25dv.c

```diff
@@ -0,0 +1,970 @@
+/*
+ * Copyright (c) 2023 Kickmaker
+ * Copyright (c) 2021 Martin Schröder <info@swedishembedded.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT st_st25dv
+
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/eeprom.h>
+#include <zephyr/drivers/eeprom/eeprom_st25dv.h>
+#include <soc.h>
+
+#define LOG_LEVEL CONFIG_EEPROM_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(eeprom_st25dv);
+
+/** @brief ST25DV registers */
+typedef enum {
+	ST25DV_REG_GPO = 0x0000,
+	ST25DV_REG_IT_TIME = 0x0001,
+	ST25DV_REG_EH_MODE = 0x0002,
+	ST25DV_REG_RF_MNGT = 0x0003,
+	ST25DV_REG_RFA1SS = 0x0004,
+	ST25DV_REG_ENDA1 = 0x0005,
+	ST25DV_REG_RFA2SS = 0x0006,
+	ST25DV_REG_ENDA2 = 0x0007,
+	ST25DV_REG_RFA3SS = 0x0008,
+	ST25DV_REG_ENDA3 = 0x0009,
+	ST25DV_REG_RFA4SS = 0x000a,
+	ST25DV_REG_I2CSS = 0x000b,
+	ST25DV_REG_LOCK_CC = 0x000c,
+	ST25DV_REG_MB_MODE = 0x000d,
+	ST25DV_REG_MB_WDG = 0x000e,
+	ST25DV_REG_LOCK_CFG = 0x000f,
+	ST25DV_REG_LOCK_DSF_ID = 0x0010,
+	ST25DV_REG_LOCK_AFI = 0x0011,
+	ST25DV_REG_DSFID = 0x0012,
+	ST25DV_REG_AFI = 0x0013,
+	ST25DV_REG_MEM_SIZE = 0x0014,
+	ST25DV_REG_BLK_SIZE = 0x0016,
+	ST25DV_REG_IC_REF = 0x0017,
+	ST25DV_REG_UUID = 0x0018,
+	ST25DV_REG_IC_REV = 0x0020,
+	ST25DV_REG_I2C_PASSWD = 0x0900,
+	ST25DV_REG_I2C_SSO_DYN = 0x2004,
+} st25dv_reg_t;
+
+/** @brief Length of the MEM_SIZE field of the ST25DV */
+#define ST25DV_MEM_SIZE_LENGTH   2u
```

Comment:
```suggestion
#define ST25DV_MEM_SIZE_LENGTH   2U
```

Please use upper-case `U` all over.